### PR TITLE
Add new method to retrieve in the same request the ids and attributes

### PR DESCRIPTION
**Pull request by belerophon**
This PR adds the interface for the new method capable of retrieving both the set of ids and attributes of a user, in a single method.

### CODE REVIEW

File: sdk/src/main/scala/eu/shiftforward/adstax/ups/api/UserProfileStorageClient.scala [hunk 1]
```diff
@@ -21,6 +21,24 @@ trait UserProfileStorageClient {
...
+   * @return a `Future` with the set of users this UserId belongs to.
+   */
+  def getIds(userId: UserId): Future[Set[UserId]]
+
+  /**
+   * Retrieves all ids associated with the provided UserId, along with its attributes.
+   *
+   * @param userId the identifier of the user we want to fetch the set of ids and attributes
+   * @return a `Future` containing a tuple with:
+   *         - the set of users this UserId belongs to
```
Comment (jcazevedo) [inline sdk/src/main/scala/eu/shiftforward/adstax/ups/api/UserProfileStorageClient.scala]: Replace `the set of users` with `the set of UserIds`.

File: sdk/src/main/scala/eu/shiftforward/adstax/ups/api/UserProfileStorageResponse.scala [hunk 2]
```diff
@@ -20,6 +20,28 @@ object GetUserResponse {
...
+
+object GetIdsResponse {
+  implicit val getIdsResponseFormat = jsonFormat1(GetIdsResponse.apply)
+}
+
+/**
+ * The response for a [[GetIdsAndData]] request.
+ *
+ * @param ids the set of ids associated with the user
+ * @param data the attributes and segment values associated with the user wrapped in a `Some` if the user exists, `None`
```
Comment (jcazevedo) [inline sdk/src/main/scala/eu/shiftforward/adstax/ups/api/UserProfileStorageResponse.scala]: Can you replace `segment` with `computed attributes` here and in the documentation of the `GetUserResponse` class?

Comment (belerophon) [reply sdk/src/main/scala/eu/shiftforward/adstax/ups/api/UserProfileStorageResponse.scala]: Corrected in a694679

File: sdk/src/main/scala/eu/shiftforward/adstax/ups/api/UserProfileStorageClient.scala (hunk 1 above)
Comment (belerophon) [reply sdk/src/main/scala/eu/shiftforward/adstax/ups/api/UserProfileStorageClient.scala]: Corrected in a694679

**Pull request merged**